File: LongestCommonPrefix.java

public class LongestCommonPrefix {

	public String longestCommonPrefix(String[] strs) {
		String output = strs[0];
		StringBuilder s = new StringBuilder();

		for (int i = 1; i < strs.length; i++) {
			for (int j = 0; j < strs[i].length() && j < output.length(); j++) {
				if (strs[i].charAt(j) == output.charAt(j)) {
					// if a common character is found, add it
					s.append(strs[i].charAt(j));
				} else {
					// break upon reaching a different character
					break;
				}
			}
			if (s.toString().length() < output.length()) {
				output = s.toString();
			}
			s.setLength(0);
		}
		return output;
	}
}

File: LongestCommonPrefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    // initialize the common prefix to be the first string
    let ans = strs[0];

    // loop through each string, comparing it with the current common prefix
    for (let i = 1; i < strs.length; i++) {
        // while the current string does not start with the common prefix
        while (strs[i].indexOf(ans) != 0) {
            // remove the last character from the common prefix
            ans = ans.substring(0, ans.length - 1);
            // if the common prefix is empty, return an empty string
            if (ans == "") {
                return "";
            }
        }
    }
    return ans;
};